### DISCUSSION TITLE
Proptypes vs defaultProps vs object default value

### DISCUSSION THREAD

**Discussion by igy95** [woowacourse/react-payments#40] [토론장]
```jsx
const ProductDetail = ({
  imgUrl = FALLBACK.PRODUCT.IMG_URL, //Error
}) => (
	// JSX...
);

ProductDetail.propTypes = {
  imgUrl: PropTypes.string.isRequired,
};

/* Not Error
ProductDetail.defaultProps = {
	imgUrl: FALLBACK.PRODUCT.IMG_URL,
} 
*/
```

props를 받아오지 않으면 parameter 에서 default value를 설정해주면 될 텐데 defaultProps, isRequired 는 왜 필요한 건가요?

**Comment by zereight** [comment]
isRequired 는 값이 들어와야함을 강제하는 용도이고,
parameter에 정의하지않고 defaultProps로 분리하는 이유는 기본값을 정하는 코드를 컴포넌트로부터 분리하기 위해서? 일까요 (다른 역할을 하는 코드이기 때문에)

**Reply by igy95** [reply]
저는 defaultProps 뿐만 아니라 isRequired 또한 컴포넌트에 명시한 인자가 아무것도 들어오지 않는 상황을 대비하기 위함이라고 생각해서 default value면 충분하다고 보았는데, 기본값과 컴포넌트를 분리한다는 시각에서 보면 또 그럴 수도 있겠네요!
